src/tests/class_tests/openms/source/MSExperiment_test.cpp

```diff
@@ -1542,6 +1542,266 @@ START_SECTION( std::ostream& operator<<(std::ostream& os, const MSExperiment& ch
 }
 END_SECTION
 
+START_SECTION((template<class MzReductionFunctionType> std::vector<std::vector<MSExperiment::CoordinateType>> aggregate(const std::vector<std::pair<RangeMZ, RangeRT>>& mz_rt_ranges, unsigned int ms_level, MzReductionFunctionType func_mz_reduction) const))
+{
+    // Create test experiment with known data
+    PeakMap exp;
+    exp.resize(4);
+    Peak1D p;
+
+    // First spectrum (MS1) at RT=1.0
+    exp[0].setRT(1.0);
+    exp[0].setMSLevel(1);
+    p.setMZ(100.0); p.setIntensity(1000.0); exp[0].push_back(p);
```

Comment:
can be done with less code as:
`exp[0] = MSSpectrum{{1.0, 29.0f}, {2.0, 60.0f}, {3.0, 34.0f}, {4.0, 29.0f}, {5.0, 37.0f}, {6.0, 31.0f}};`